test/integration/h2_fuzz.cc

```diff
@@ -0,0 +1,215 @@
+    ENVOY_LOG_MISC(trace, "Sending continuation frame");
+    h2_frame = Http2Frame::makeEmptyContinuationFrame(stream_idx, headers_flags);
+  } break;
+  case test::integration::H2TestFrame::kData: {
+    const Http2Frame::DataFlags data_flags =
+        static_cast<Http2Frame::DataFlags>(proto_frame.data().flags());
+    uint32_t stream_idx = proto_frame.data().stream_index();
+    ENVOY_LOG_MISC(trace, "Sending data frame");
+    h2_frame = Http2Frame::makeEmptyDataFrame(stream_idx, data_flags);
+  } break;
+  case test::integration::H2TestFrame::kPriority: {
+    uint32_t stream_idx = proto_frame.priority().stream_index();
+    uint32_t dependent_idx = proto_frame.priority().dependent_index();
+    ENVOY_LOG_MISC(trace, "Sending priority frame");
+    h2_frame = Http2Frame::makePriorityFrame(stream_idx, dependent_idx);
+  } break;
+  case test::integration::H2TestFrame::kPushPromise: {
+    const Http2Frame::HeadersFlags headers_flags =
+        static_cast<Http2Frame::HeadersFlags>(proto_frame.push_promise().flags());
+    uint32_t stream_idx = proto_frame.push_promise().stream_index();
+    uint32_t promised_stream_idx = proto_frame.push_promise().promised_stream_index();
+    ENVOY_LOG_MISC(trace, "Sending push promise frame");
+    h2_frame =
+        Http2Frame::makeEmptyPushPromiseFrame(stream_idx, promised_stream_idx, headers_flags);
+  } break;
+  case test::integration::H2TestFrame::kResetStream: {
+    uint32_t stream_idx = proto_frame.reset_stream().stream_index();
+    const Http2Frame::ErrorCode error_code =
+        static_cast<Http2Frame::ErrorCode>(proto_frame.reset_stream().error_code());
+    ENVOY_LOG_MISC(trace, "Sending reset stream frame");
+    h2_frame = Http2Frame::makeResetStreamFrame(stream_idx, error_code);
+  } break;
+  case test::integration::H2TestFrame::kGoAway: {
+    uint32_t last_stream_idx = proto_frame.go_away().last_stream_index();
+    const Http2Frame::ErrorCode error_code =
+        static_cast<Http2Frame::ErrorCode>(proto_frame.go_away().error_code());
+    ENVOY_LOG_MISC(trace, "Sending go-away frame");
+    h2_frame = Http2Frame::makeEmptyGoAwayFrame(last_stream_idx, error_code);
+  } break;
+  case test::integration::H2TestFrame::kWindowUpdate: {
+    uint32_t stream_idx = proto_frame.window_update().stream_index();
+    uint32_t increment = proto_frame.window_update().increment();
+    ENVOY_LOG_MISC(trace, "Sending windows_update frame");
+    h2_frame = Http2Frame::makeWindowUpdateFrame(stream_idx, increment);
+  } break;
+  case test::integration::H2TestFrame::kMalformedRequest: {
+    uint32_t stream_idx = proto_frame.malformed_request().stream_index();
+    ENVOY_LOG_MISC(trace, "Sending malformed_request frame");
+    h2_frame = Http2Frame::makeMalformedRequest(stream_idx);
+  } break;
+  case test::integration::H2TestFrame::kMalformedRequestWithZerolenHeader: {
+    uint32_t stream_idx = proto_frame.malformed_request_with_zerolen_header().stream_index();
```

Comment:
Nit: I'd make most of these temp vars `const`.

Reply:
Done.